python/sglang/srt/function_call_parser.py

```diff
@@ -445,31 +469,37 @@ def has_tool_call(self, text: str) -> bool:
 
     def detect_and_parse(self, text: str, tools: List[Tool]) -> StreamingParseResult:
         """Parse function calls from text, handling multiple JSON objects."""
-        if "<|python_tag|>" not in text and not text.startswith("{"):
-            return StreamingParseResult(normal_text=text, calls=[])
-
-        if "<|python_tag|>" in text:
-            normal_text, action_text = text.split("<|python_tag|>")
-        else:
-            normal_text, action_text = "", text
+        try:
+            if "<|python_tag|>" not in text and not text.startswith("{"):
+                return StreamingParseResult(normal_text=text, calls=[])
 
-        # Split by semicolon and process each part
-        json_parts = [part.strip() for part in action_text.split(";") if part.strip()]
-        all_actions = []
-        for part in json_parts:
-            try:
-                # Parse each individual JSON object
-                action = json.loads(part)
-                all_actions.append(action)
-            except json.JSONDecodeError as e:
-                logger.warning(f"Failed to parse JSON part: {part}")
-                logger.warning(f"JSON parse error: {str(e)}")
-                continue
-        calls = []
-        # Only process if we found valid JSON objects
-        if all_actions:
-            calls = self.parse_base_json(all_actions, tools)
-        return StreamingParseResult(normal_text=normal_text, calls=calls)
+            if "<|python_tag|>" in text:
+                normal_text, action_text = text.split("<|python_tag|>")
+            else:
+                normal_text, action_text = "", text
+
+            # Split by semicolon and process each part
+            json_parts = [
+                part.strip() for part in action_text.split(";") if part.strip()
+            ]
+            all_actions = []
+            for part in json_parts:
+                try:
+                    # Parse each individual JSON object
+                    action = json.loads(part)
+                    all_actions.append(action)
+                except json.JSONDecodeError as e:
+                    logger.warning(f"Failed to parse JSON part: {part}")
+                    logger.warning(f"JSON parse error: {str(e)}")
+                    continue
+            calls = []
+            # Only process if we found valid JSON objects
+            if all_actions:
+                calls = self.parse_base_json(all_actions, tools)
+            return StreamingParseResult(normal_text=normal_text, calls=calls)
+        except Exception as e:
```

Comment:
do we need to catch exception here? Feels the exception mattered has already been catched under https://github.com/sgl-project/sglang/blob/6ad0fa3c829b92d7c37ace0ffaa2cbe17190b09c/python/sglang/srt/function_call_parser.py#L491-L494